### PR TITLE
WDFN-103 Multiple time-series per parameter code

### PR DESCRIPTION
**Pull request by danielnaab**

### CODE REVIEW

File: assets/src/scripts/components/hydrograph/store.js [hunk 1]
```diff
@@ -1,43 +1,57 @@
+const merge = require('lodash/merge');
```
Comment (mbucknell) [inline assets/src/scripts/components/hydrograph/store.js]: I don't think this got added to package.json.

Comment (danielnaab) [reply assets/src/scripts/components/hydrograph/store.js]: Oh, must be a dependency of another package we use. Will add.

File: assets/src/scripts/components/hydrograph/schema.js [hunk 2]
```diff
@@ -0,0 +1,130 @@
...
+            variableName: replaceHtmlEntities(variable.variableName),
+            variableCode: variable.variableCode[0]
+        };
+    }
+});
+
+// timeSeries schema
+const qualifier = new schema.Entity('qualifiers', {}, {idAttribute: 'qualifierCode'});
+const method = new schema.Entity('methods', {}, {idAttribute: 'methodID'});
+const timeSeries = tsKey => new schema.Entity('timeSeries', {
```
Comment (mbucknell) [inline assets/src/scripts/components/hydrograph/schema.js]: Style issue: I don't think I like using arrow functions when defining a function object. Is there a reason you do so?

Comment (danielnaab) [reply assets/src/scripts/components/hydrograph/schema.js]: Just conciseness. But looking now, those should all perhaps be memoized..

Comment (mbucknell) [reply assets/src/scripts/components/hydrograph/schema.js]: That's a great idea!

File: assets/src/scripts/components/hydrograph/parameters.js [hunk 3]
```diff
@@ -129,12 +131,21 @@ export const plotSeriesSelectTable = function (elem, {availableTimeseries, layou
                 // under the appropriate column headers
                 if (!screenSizeCheck) {
                     tr.append('td')
-                        .html(parm => parm[1].currentYear ? '<i class="fa fa-check" aria-label="Current year data available"></i>' : '');
+                        .html(parm => {
+                            const subScript = parm[1].currentYear > 1 ? `<sub>${parm[1].currentYear}</sub>` : '';
```
Comment (mbucknell) [inline assets/src/scripts/components/hydrograph/parameters.js]: Might be more concise if we have a local function which creates the html. We also lose the subscripts in the tooltip

File: assets/src/scripts/components/hydrograph/index.js [hunk 4]
```diff
@@ -129,32 +142,37 @@ const plotSvgDefs = function(elem) {
...
 
 
-const plotLegend = function(elem, {displayItems, layout}) {
-    elem.select('.legend').remove();
-    let plotMarkers = createLegendMarkers(displayItems);
-    drawSimpleLegend(elem, plotMarkers, layout);
+const timeSeriesLegend = function(elem) {
+    elem.append('div')
+        .attr('class', 'hydrograph-container')
+        .append('svg')
```
Comment (mbucknell) [inline assets/src/scripts/components/hydrograph/index.js]: Why change to drawing the legend in a separate SVG?

File: assets/src/scripts/components/hydrograph/index.js [hunk 5]
```diff
@@ -294,4 +363,4 @@ const attachToNode = function (node, {siteno} = {}) {
 };
 
 
-module.exports = {attachToNode, timeSeriesGraph};
+module.exports = {attachToNode, timeSeriesLegend, timeSeriesGraph};
```
Comment (mbucknell) [inline assets/src/scripts/components/hydrograph/index.js]: I'm not sure I like that we are exporting timeSeriesLegend and timeSeriesGraph only for testing. What do others think? I think they can be tested particularly now that we have separated the state from the rendering without exporting these functions.

File: assets/src/scripts/components/hydrograph/index.js (hunk 4 above)
Comment (mbucknell) [reply assets/src/scripts/components/hydrograph/index.js]: We will need to add at least some of the accessibility features to it such as a title and description.

File: assets/src/scripts/components/hydrograph/parameters.js (hunk 3 above)
Comment (mbucknell) [reply assets/src/scripts/components/hydrograph/parameters.js]: That's not what I meant. We could have a local function that creates the checkmark with the appropriate subscript.

File: assets/src/scripts/components/hydrograph/index.js (hunk 4 above)
Comment (danielnaab) [reply assets/src/scripts/components/hydrograph/index.js]: Yes, missed that.

File: assets/src/scripts/components/hydrograph/parameters.js (hunk 3 above)
Comment (danielnaab) [reply assets/src/scripts/components/hydrograph/parameters.js]: I caught that - but I was wondering what you meant by the reference to tooltip

Comment (mbucknell) [reply assets/src/scripts/components/hydrograph/parameters.js]: Must have missed it. The subscripts are on the small screen tooltip.

Comment (danielnaab) [reply assets/src/scripts/components/hydrograph/parameters.js]: Ah, I see. Yep, one more thing..

**Pull request merged**